File: python/leetcode228.py
class Solution:
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        if len(nums) == 0:
            return []
        if len(nums) == 1:
            return [str(nums[0])]
        res = []
        low = nums[0]
        for i in range(1,len(nums)):
            if nums[i] - nums[i-1] > 1:
                high = nums[i-1]
                if low == high:
                    res.append(str(low))
                else:
                    res.append(str(low)+"->"+str(high))
                low = nums[i]

        if low < nums[-1]:
            res.append(str(low) + "->" + str(nums[-1]))
        else:
            res.append(str(low))
        return res

File: leetcodec++/leetcode228.cpp
//
// Created by 黄敏辉 on 2019-09-06.
//

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> res;
        int i = 0;
        while (i < nums.size()){
            int k = i;
            while (i + 1 < nums.size() && nums[i] + 1 == nums[i+1])
                i++;
            res.push_back(to_s(nums[k],nums[i]));
            i++;
        }
        return res;
    }

    string to_s(int i, int j){
        if (i == j)
            return to_string(i);
        else
            return to_string(i) + "->" + to_string(j);
    }
};